packages/testcontainers/src/docker.ts

```diff
@@ -0,0 +1,192 @@
+import Dockerode, {DockerOptions, Container} from 'dockerode'
+import fetch from 'node-fetch'
+import JSONBig from 'json-bigint'
+
+export type Network = 'mainnet' | 'testnet' | 'regtest'
+
+export interface StartOptions {
+  // TODO(fuxingloh): change to cookie based auth soon
+  user: string
+  password: string
+}
+
+/**
+ * DeFiChain defid node managed in docker
+ */
+export abstract class DeFiChainDocker {
+  protected static readonly PREFIX = 'defichain-testcontainers-'
+  protected readonly docker: Dockerode
+  protected readonly image = 'defi/defichain:1.5.0'
+  protected readonly network: Network
+
+  protected container?: Container
+  protected startOptions?: StartOptions
+
+  protected constructor(network: Network, options?: DockerOptions) {
+    this.docker = new Dockerode(options)
+    this.network = network
+  }
+
+  /**
+   * Generate a name for a new docker container with network type and random number
+   */
+  protected generateName(): string {
+    const rand = Math.floor(Math.random() * 10000000)
+    return `${DeFiChainDocker.PREFIX}-${this.network}-${rand}`
+  }
+
+  protected getCmd(opts: StartOptions): string[] {
+    return [
```

Comment:
I think this function might need to accept some parameters. Like if they need to do `-reindex` or maybe if they can pass their config file like in the app `-conf`. You can ignore this if you're planning to do it in the future.

Reply:
I am using the polymorphic construct for this, it can be adapted as such

```ts
protected getCmd(opts: StartOptions): string[] {
  return [...super.getCmd(opts), '-regtest=1']
}
```